src/api/app/controllers/webui/projects/project_configuration_controller.rb

```diff
@@ -0,0 +1,37 @@
+module Webui
+  module Projects
+    class ProjectConfigurationController < WebuiController
+      before_action :set_project
+
+      def show
+        sliced_params = params.slice(:rev)
+        sliced_params.permit!
+        @content = @project.config.content(sliced_params.to_h)
+        switch_to_webui2
+        return if @content
+        flash[:error] = @project.config.errors.full_messages.to_sentence
+        redirect_to projects_path(nextstatus: 404)
+      end
+
+      def update
+        authorize @project, :update?
+        content = ::ProjectConfigurationService::ProjectConfigurationUpdater.new(@project, User.current, params).call
```

Comment:
This is not the content but the result. In general I would prefer to not rely on side effects of the Updater class (returning the result but expecting that it does change the @project object). We could return self and ask self about the result and the errors.

```
    def call
       @params[:user] = @user.login
       @saved = @project.config.save(sliced_params, @params[:config])
       self
     end

    def saved?
        @saved || false
    end

    def errors
        @project.config.errors.full_messages.to_sentence
    end
```

Then you can do:

```
result = ::ProjectConfigurationService::ProjectConfigurationUpdater.new(@project, User.current, params).call

if result.saved?
  ....
  flash.now[:error] = result.errors
else

end
```